packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -171,13 +197,61 @@ class BuildIOSArchiveCommand extends _BuildIOSSubCommand {
       LineSplitter.split(result.stderr)
           .where((String line) => line.contains('error: '))
           .forEach(errorMessage.writeln);
-      throwToolExit('Encountered error while building IPA:\n$errorMessage');
+
+      globals.printError('Encountered error while creating the IPA:');
+      globals.printError(errorMessage.toString());
+      globals.printError('Try distributing the app in Xcode: "open $archivePath"');
+      throwToolExit(null);
     }
 
-    globals.printStatus('Built IPA to $outputPath.');
+    globals.printStatus('Built IPA to $absoluteOutputPath.');
+
+    if (isAppStoreUpload) {
+      globals.printStatus('To upload to the App Store either:');
+      globals.printStatus(
+        '1. Drag and drop the "$relativeOutputPath/*.ipa" bundle into the Apple Transport macOS app https://apps.apple.com/us/app/transporter/id1450874784',
+        indent: 4,
+      );
+      globals.printStatus(
+        '2. Run "xcrun altool --upload-app --type ios -f $relativeOutputPath/*.ipa --apiKey your_api_key --apiIssuer your_issuer_id".',
+        indent: 4,
+      );
+      globals.printStatus(
+        'See "man altool" for details about how to authenticate with the App Store Connect API key.',
+        indent: 7,
+      );
+    }
 
     return FlutterCommandResult.success();
   }
+
+  File _createExportPlist() {
+    // Create the plist to be passed into xcodebuild -exportOptionsPlist.
+    final StringBuffer plistContents = StringBuffer('''
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+    <dict>
+        <key>method</key>
+''');
+
+    plistContents.write('''
+        <string>${stringArg('export-method')}</string>
+    ''');
+    // Bitcode is off by default in Flutter iOS apps.
+    plistContents.write('''
+    <key>uploadBitcode</key>
+        <false/>
```

Comment:
See https://github.com/flutter/flutter/issues/39356 and https://github.com/flutter/flutter/pull/73607#issuecomment-760591579 for bitcode context.

Unfortunately it didn't figure it out based on the Xcode build setting and defaulted to `"uploadBitcode" => 1`, kind of weird it would be set differently during distribution.  Flutter's template has it off by default and most developers don't change the defaults.

But instead of assuming, I'll read `ENABLE_BITCODE` off the build setting.